### PR TITLE
Re-organize fetcher directory structure

### PR DESCRIPTION
**Pull request by orestisfl**
### Summary of your changes

Mostly moving stuff around

Follow a more structured hierarchy instead of having 4 fetching-related directories under `resources/`.

Now, we have `resources/fetching` and under it we have:
- `fetchers/` with the specific implementation of each fetcher
- `registry/`
- `manager/`
- `condition/`

Some names have been modified to be more friendly to go conventions

Follow-up from https://github.com/elastic/cloudbeat/pull/1033

### Related Issues
- Related: https://github.com/elastic/cloudbeat/issues/756

### Checklist
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

**Comment by oren-zohar** [comment]
- related: https://github.com/elastic/cloudbeat/issues/664

File: resources/fetching/condition/is_leader.go [hunk 1]
```diff
@@ -15,30 +15,30 @@
...
+package condition
 
 import (
 	"github.com/elastic/cloudbeat/resources/fetching"
 	"github.com/elastic/cloudbeat/uniqueness"
 	"github.com/elastic/elastic-agent-libs/logp"
 )
 
-type LeaseFetcherCondition struct {
+type IsLeader struct {
```
Comment (uri-weisman) [inline resources/fetching/condition/is_leader.go:26]: Is it according to Go conventions? the previous name is better in my opinion.

File: resources/fetching/manager/manager.go [hunk 2]
```diff
@@ -90,14 +90,14 @@ func (m *Manager) fetchAndSleep(ctx context.Context) {
...
-	m.log.Infof("Manager triggered fetching for %d fetchers", len(m.fetchers.Keys()))
+	m.log.Infof("Manager triggered fetching for %d fetchers", len(m.fetcherRegistry.Keys()))
 
 	start := time.Now()
 
 	seq := time.Now().Unix()
 	m.log.Infof("Cycle %d has started", seq)
 	wg := &sync.WaitGroup{}
-	for _, key := range m.fetchers.Keys() {
+	for _, key := range m.fetcherRegistry.Keys() {
```
Comment (uri-weisman) [inline resources/fetching/manager/manager.go:100]: 👍

**Review by uri-weisman — APPROVED** [review]
Great intiative @orestisfl!
I seems that all mocks under the fetching folder are not used, probably they are being auto-generated by mockery and we need to figure out how to only create mocks for interfaces and not types.

File: resources/fetching/condition/is_leader.go (hunk 1 above)
Comment (orestisfl) [reply resources/fetching/condition/is_leader.go:26]: I believe that the name `condition.IsLeader` is easier and more understandable than `condition.LeaseFetcherCondition`. Happy to rename though if you prefer it.

Comment (uri-weisman) [reply resources/fetching/condition/is_leader.go:26]: Thanks, Let's keep it.

**Pull request merged**